insights/cleaner/filters.py

```diff
@@ -32,3 +32,26 @@ def parse_line(self, line, **kwargs):
 
     def generate_report(self, report_dir, archive_name):
         pass  # pragma: no cover
+
+    @staticmethod
+    def filter_content(lines, allowlist):
+        """
+        Filter content based on allowlist.
+        When a key of allowlist is found in a line, it is added to the result
+        list. The key is removed from the allowlist when enough lines
+        containing the key were found.
+        When the allowlist is empty, an empty result is returned.
+
+        :param lines: list of lines
+        :param allowlist: dictionary of allowlist
+        :return: list of lines
+        """
+        result = []
+        for line in lines:
+            for a_key in list(allowlist.keys()):  # copy keys to avoid RuntimeError
+                if a_key in line:
+                    allowlist[a_key] -= 1
+                    # stop checking it when enough lines contain the key were found
+                    allowlist.pop(a_key) if allowlist[a_key] == 0 else None
+                    result.append(line)
```

Comment:
Seems we would need just one time of this `result.append(line)` be called for each content line, right?